File: 0219 contains-duplicate-ii/solution.go
func containsNearbyDuplicate(nums []int, k int) bool {
    set:=make(map[int]bool)

    for i,v := range nums {
        if i>k {
            delete(set,nums[i-k-1])
        }

        
        if set[v] {
            return true
        }

        set[v]=true
    }
    return false

}

File: 0219 contains-duplicate-ii/solution.rs
use std::collections::HashSet;
impl Solution {
    pub fn contains_nearby_duplicate(nums: Vec<i32>, k: i32) -> bool {
        let mut set: HashSet<i32> = HashSet::new();
        let k=k as usize;
        for i in 0..nums.len() {
            if i>k {
                set.remove(&nums[i-k-1]);
            }

            if !set.insert(nums[i]) {
                return true;
            }
        }
        false
    }
}
